canonical/react-components
# fix: contextual menu container overflow

## Done

This pull request fixes the menu position when scrolling in overflowing containers.

It also simplifies the implementation by moving the dropdown element to the container.
This way, we're able to rely on CSS positioning for the menu and do not have to calculate it manually using JavaScript.

It also adds isEnabled options to existing hooks to be able to use them as a replacement for the logic that was provided by use-portal.

- fix contextual menu dropdown not being scrolled correctly in overflowing containers
- replace deprecated ClientRect with DOMRect
- fix dropdown props being passed incorrectly to the wrapper
- simplify implementation by removing the Portal and replacing it with useState
- add options to useOnEscapePressed and useClickOutside
- rename useClickOutside to useOnClickOutside for consistency
- remove the portal ref hack

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Go to the overflow example
- Ensure that once you open the dropdown and scroll, menu position is in sync with the toggle
- Ensure you can switch autoAdjust on and off
- When enabled, Dropdown menu position should be adjusted in case of a long menu when position is set to the left.

## Fixes

Fixes: https://warthogs.atlassian.net/browse/MAASENG-2407
